### ISSUE TITLE
Additional examples for Stream operator docs

### ISSUE THREAD

**Issue by raboof** [akka/akka#25468]
Many operators in the stream operator documentation section at https://doc.akka.io/docs/akka/current/stream/operators/index.html do not yet have code examples. Adding such a code example is a great contribution.

An example of an operator that does have example code is https://doc.akka.io/docs/akka/current/stream/operators/ActorFlow/ask.html#examples . 

Example code in Akka is included in the source tree and is compiled (and thus typechecked) on every build. Sometimes the examples even come from unit tests, so they are not only compiled but even ran and validated. If you can add or find a test that can be used as documentation that is great, otherwise the code can be added to the akka-docs subproject.

To add an example:
* Optionally, comment in this issue to 'claim' an operator
* Add (or find) example code, mark it as a Paradox snippet (with `//#xxx` comments)
* Add a snippet to the markdown file of your operator to include that snippet

**Comment by patriknw** [comment]
I agree, this is an excellent community ticket and we really need the help to complete all of it.

**Comment by ch33hau** [comment]
Hi @raboof @patriknw , I am interested in working with this task.

I would like to start with: `fold`, `foldAsync`, `scanAsync`, `recover`, `recoverWith`, `grouped` :)

**Comment by patriknw** [comment]
@ch33hau That would be excellent. `fold` is in progress in https://github.com/akka/akka/pull/27850, but others are free for grab.

**Comment by johanandren** [comment]
Picking up `Source.unfold` and `Source.unfoldAsync`

**Comment by chbatey** [comment]
Picking up take/takeWhile

**Comment by chbatey** [comment]
Picking up unfoldResource and unfoldResourceAsync

**Comment by ignasi35** [comment]
I'm doing `completionStageSource`.

**Comment by ignasi35** [comment]
I'm looking at `{as,from}JavaStream` converters.

**Comment by JoelWee** [comment]
Hi, I would like to pick up mergePreferred and mergePrioritized :)

**Comment by mariojaros** [comment]
Hi everyone, in my first contribution I can take onComplete.

**Comment by muskan3006** [comment]
Hi everyone, this is my first contribution so I can take fold for Sink

**Comment by muskan3006** [comment]
I will take reduce from simple operators

**Comment by muskan3006** [comment]
Also, there are some operators that are in the docs but not mentioned here

**Comment by ignasi35** [comment]
> Also, there are some operators that are in the docs but not mentioned here

You have a keen eye, @muskan3006 :)

The list at the top of this issue may be out of date. Any inconsistency you detect, feel free to report and we'll update the description.

**Comment by muskan3006** [comment]
> > Also, there are some operators that are in the docs but not mentioned here
> 
> You have a keen eye, @muskan3006 :)
> 
> The list at the top of this issue may be out of date. Any inconsistency you detect, feel free to report and we'll update the description.

Sure, I'll mention them here

**Comment by muskan3006** [comment]
Hi, I'll take asSourceWithContext from Source operators

**Comment by muskan3006** [comment]
collection operator is not mentioned here

**Comment by muskan3006** [comment]
Hi I will take completionTimeout from Time aware operators

**Comment by muskan3006** [comment]
Hi I will take headoption

**Comment by muskan3006** [comment]
I'll take cancelled

**Comment by nitikagarw** [comment]
Hi, Can I please take `Seq` Sink Operator?

**Comment by patriknw** [comment]
@nitikagarw That would be great, thanks.

**Comment by nitikagarw** [comment]
@patriknw `Seq` operator is already done: https://github.com/akka/akka/blob/master/akka-docs/src/main/paradox/stream/operators/Sink/seq.md#example

Can I please pick `asInputStream` and `asOutputStream` from the Converters list?

**Comment by johanandren** [comment]
Sounds good, please go ahead @nitikagarw

**Comment by mdipirro** [comment]
Hello, I'll take `groupBy` from Nesting and flattening operators.

**Comment by mdipirro** [comment]
Hi, I'll take `watchTermination` as well.

**Comment by nitikagarw** [comment]
Hi, Can I please pick `foldAsync` and `mapError` from the list of Simple Operators?

**Comment by johanandren** [comment]
Please go ahead @nitikagarw !

**Comment by muskan3006** [comment]
Hi, I'll take asPublisher from SInk operators

**Comment by muskan3006** [comment]
Hi, picking fromSubscriber

**Comment by marijkhan** [comment]
Hi, is this still open for contribution? Can I pick `fromMaterializer` from the list of simple operators?

**Comment by patriknw** [comment]
absolutely, thanks

**Comment by wolf8334** [comment]
I noticed that the tab before lazyFlow is grayed out, so I assumed it was an unclaimed task. However, after checking the code, I found that it already has markdown documentation and code examples. How can I find operators that still need documentation examples?

**Comment by johanandren** [comment]
I've gone through and checked off the ones in this list that has examples, so you can chose an unchecked in the list, but I think we have also added plenty operators since this list was created, so if you find an operator page in https://doc.akka.io/libraries/akka-core/current/stream/operators/index.html#asynchronous-operators that doesn't have an example and want to contribute one for it, that is welcome.

**Comment by wolf8334** [comment]
I'd like to contribute an example for StreamConverters.javaCollector. I'll follow the format from PR #28746. If this goes well, I'd like to continue contributing more examples.
